After testing, there are still issues with deserialization.
Caused by: java.lang.ArrayIndexOutOfBoundsException: 49
at org.apache.fury.resolver.MetaStringBytes.decode(MetaStringBytes.java:69) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.populateBytesToClassInfo(ClassResolver.java:1708) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.loadBytesToClassInfo(ClassResolver.java:1699) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.readClassInfoFromBytes(ClassResolver.java:1686) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.readClassInfo(ClassResolver.java:1608) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.readRef(Fury.java:857) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.deserializeJavaObjectAndClass(Fury.java:1225) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.deserializeJavaObjectAndClass(Fury.java:1245) ~[fury-core-0.7.0.jar!/:0.7.0]

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/BaseFurySerialization.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/BaseFurySerialization.java
@@ -17,10 +17,6 @@
 
 package org.apache.dubbo.common.serialize.fury.dubbo;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.util.Optional;
 import org.apache.dubbo.common.URL;
 import org.apache.dubbo.common.serialize.ObjectInput;
 import org.apache.dubbo.common.serialize.ObjectOutput;
@@ -31,6 +27,11 @@ import org.apache.fury.Fury;
 import org.apache.fury.collection.Tuple2;
 import org.apache.fury.memory.MemoryBuffer;
 import org.apache.fury.util.LoaderBinding;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Optional;
 
 /**
  * Fury serialization framework integration with dubbo.

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryCheckerListener.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryCheckerListener.java
@@ -17,8 +17,6 @@
 
 package org.apache.dubbo.common.serialize.fury.dubbo;
 
-import java.io.Serializable;
-import java.util.Set;
 import org.apache.dubbo.common.utils.AllowClassNotifyListener;
 import org.apache.dubbo.common.utils.SerializeCheckStatus;
 import org.apache.dubbo.common.utils.SerializeSecurityManager;
@@ -29,6 +27,9 @@ import org.apache.fury.exception.InsecureException;
 import org.apache.fury.resolver.AllowListChecker;
 import org.apache.fury.serializer.Serializer;
 import org.apache.fury.serializer.SerializerFactory;
+
+import java.io.Serializable;
+import java.util.Set;
 
 @SuppressWarnings("rawtypes")
 public class FuryCheckerListener implements AllowClassNotifyListener, SerializerFactory {

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryCheckerListener.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryCheckerListener.java
@@ -46,10 +46,10 @@ public class FuryCheckerListener implements AllowClassNotifyListener, Serializer
   @Override
   public void notifyPrefix(Set<String> allowedList, Set<String> disAllowedList) {
     for (String prefix : allowedList) {
-      checker.allowClass(prefix);
+      checker.allowClass(prefix + "*");
     }
     for (String prefix : disAllowedList) {
-      checker.disallowClass(prefix);
+      checker.disallowClass(prefix + "*");
     }
   }
 
@@ -90,4 +90,5 @@ public class FuryCheckerListener implements AllowClassNotifyListener, Serializer
     }
     return null;
   }
+
 }

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java
@@ -23,6 +23,7 @@ import java.lang.reflect.Type;
 import org.apache.dubbo.common.serialize.ObjectInput;
 
 import org.apache.fury.Fury;
+import org.apache.fury.io.BlockedStreamUtils;
 import org.apache.fury.io.FuryInputStream;
 import org.apache.fury.memory.MemoryBuffer;
 
@@ -40,7 +41,7 @@ public class FuryObjectInput implements ObjectInput {
 
   @Override
   public Object readObject() {
-    return fury.deserializeJavaObjectAndClass(new FuryInputStream(input, buffer.size()));
+    return BlockedStreamUtils.deserialize(fury, input);
   }
 
   @Override

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java
@@ -17,110 +17,110 @@
 
 package org.apache.dubbo.common.serialize.fury.dubbo;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.lang.reflect.Type;
 import org.apache.dubbo.common.serialize.ObjectInput;
 
 import org.apache.fury.Fury;
 import org.apache.fury.io.BlockedStreamUtils;
-import org.apache.fury.io.FuryInputStream;
 import org.apache.fury.memory.MemoryBuffer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
 
 @SuppressWarnings("unchecked")
 public class FuryObjectInput implements ObjectInput {
-  private final Fury fury;
-  private final MemoryBuffer buffer;
-  private final InputStream input;
+    private final Fury fury;
+    private final MemoryBuffer buffer;
+    private final InputStream input;
 
-  public FuryObjectInput(Fury fury, MemoryBuffer buffer, InputStream input) {
-    this.fury = fury;
-    this.buffer = buffer;
-    this.input = input;
-  }
-
-  @Override
-  public Object readObject() {
-    return BlockedStreamUtils.deserialize(fury, input);
-  }
-
-  @Override
-  public <T> T readObject(Class<T> cls) {
-    return (T) readObject();
-  }
-
-  @Override
-  public <T> T readObject(Class<T> cls, Type type) {
-    return (T) readObject();
-  }
-
-  @Override
-  public boolean readBool() throws IOException {
-    readBytes(buffer.getHeapMemory(), 1);
-    return buffer.getBoolean(0);
-  }
-
-  @Override
-  public byte readByte() throws IOException {
-    readBytes(buffer.getHeapMemory(), 1);
-    return buffer.getByte(0);
-  }
-
-  @Override
-  public short readShort() throws IOException {
-    readBytes(buffer.getHeapMemory(), 2);
-    return buffer.getInt16(0);
-  }
-
-  @Override
-  public int readInt() throws IOException {
-    readBytes(buffer.getHeapMemory(), 4);
-    return buffer.getInt32(0);
-  }
-
-  @Override
-  public long readLong() throws IOException {
-    readBytes(buffer.getHeapMemory(), 8);
-    return buffer.getInt64(0);
-  }
-
-  @Override
-  public float readFloat() throws IOException {
-    readBytes(buffer.getHeapMemory(), 4);
-    return buffer.getFloat32(0);
-  }
-
-  @Override
-  public double readDouble() throws IOException {
-    readBytes(buffer.getHeapMemory(), 8);
-    return buffer.getFloat64(0);
-  }
-
-  @Override
-  public String readUTF() throws IOException {
-    int size = readInt();
-    buffer.readerIndex(0);
-    buffer.ensure(size);
-    readBytes(buffer.getHeapMemory(), size);
-    if (buffer.readBoolean()) {
-      return fury.readJavaString(buffer);
-    } else {
-      return null;
+    public FuryObjectInput(Fury fury, MemoryBuffer buffer, InputStream input) {
+        this.fury = fury;
+        this.buffer = buffer;
+        this.input = input;
     }
-  }
 
-  @Override
-  public byte[] readBytes() throws IOException {
-    int size = readInt();
-    byte[] bytes = new byte[size];
-    readBytes(bytes, size);
-    return bytes;
-  }
-
-  private void readBytes(byte[] bytes, int size) throws IOException {
-    int off = 0;
-    while (off != size) {
-      off += input.read(bytes, off, size - off);
+    @Override
+    public Object readObject() {
+        return BlockedStreamUtils.deserialize(fury, input);
     }
-  }
+
+    @Override
+    public <T> T readObject(Class<T> cls) {
+        return (T) readObject();
+    }
+
+    @Override
+    public <T> T readObject(Class<T> cls, Type type) {
+        return (T) readObject();
+    }
+
+    @Override
+    public boolean readBool() throws IOException {
+        readBytes(buffer.getHeapMemory(), 1);
+        return buffer.getBoolean(0);
+    }
+
+    @Override
+    public byte readByte() throws IOException {
+        readBytes(buffer.getHeapMemory(), 1);
+        return buffer.getByte(0);
+    }
+
+    @Override
+    public short readShort() throws IOException {
+        readBytes(buffer.getHeapMemory(), 2);
+        return buffer.getInt16(0);
+    }
+
+    @Override
+    public int readInt() throws IOException {
+        readBytes(buffer.getHeapMemory(), 4);
+        return buffer.getInt32(0);
+    }
+
+    @Override
+    public long readLong() throws IOException {
+        readBytes(buffer.getHeapMemory(), 8);
+        return buffer.getInt64(0);
+    }
+
+    @Override
+    public float readFloat() throws IOException {
+        readBytes(buffer.getHeapMemory(), 4);
+        return buffer.getFloat32(0);
+    }
+
+    @Override
+    public double readDouble() throws IOException {
+        readBytes(buffer.getHeapMemory(), 8);
+        return buffer.getFloat64(0);
+    }
+
+    @Override
+    public String readUTF() throws IOException {
+        int size = readInt();
+        buffer.readerIndex(0);
+        buffer.ensure(size);
+        readBytes(buffer.getHeapMemory(), size);
+        if (buffer.readBoolean()) {
+            return fury.readJavaString(buffer);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public byte[] readBytes() throws IOException {
+        int size = readInt();
+        byte[] bytes = new byte[size];
+        readBytes(bytes, size);
+        return bytes;
+    }
+
+    private void readBytes(byte[] bytes, int size) throws IOException {
+        int off = 0;
+        while (off != size) {
+            off += input.read(bytes, off, size - off);
+        }
+    }
 }

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectOutput.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectOutput.java
@@ -17,13 +17,14 @@
 
 package org.apache.dubbo.common.serialize.fury.dubbo;
 
-import java.io.IOException;
-import java.io.OutputStream;
 import org.apache.dubbo.common.serialize.ObjectOutput;
 
 import org.apache.fury.Fury;
 import org.apache.fury.io.BlockedStreamUtils;
 import org.apache.fury.memory.MemoryBuffer;
+
+import java.io.IOException;
+import java.io.OutputStream;
 
 /**
  * Fury implementation for {@link ObjectOutput}.

--- a/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectOutput.java
+++ b/dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectOutput.java
@@ -22,6 +22,7 @@ import java.io.OutputStream;
 import org.apache.dubbo.common.serialize.ObjectOutput;
 
 import org.apache.fury.Fury;
+import org.apache.fury.io.BlockedStreamUtils;
 import org.apache.fury.memory.MemoryBuffer;
 
 /**
@@ -41,7 +42,7 @@ public class FuryObjectOutput implements ObjectOutput {
   }
 
   public void writeObject(Object obj) {
-    fury.serializeJavaObjectAndClass(output, obj);
+      BlockedStreamUtils.serialize(fury, output, obj);
   }
 
   public void writeBool(boolean v) throws IOException {

